In [57]:
import pandas as pd
import seaborn as sns

df = sns.load_dataset("titanic")

df.shape


(891, 15)

891 lignes et 15 colonnes

In [58]:
df.dtypes

survived          int64
pclass            int64
sex                 str
age             float64
sibsp             int64
parch             int64
fare            float64
embarked            str
class          category
who                 str
adult_male         bool
deck           category
embark_town         str
alive               str
alone              bool
dtype: object

Voici le type de chaque colonne.

In [59]:
df.head()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True


On peut constater que les colonnes:
"survived et "alive"
"sex", "who" "adult_male"
"pclass" et "class"
"embarked" et "embarked_town"
"sibsp" et "alone" sont liées


In [60]:
df.isna().sum()

survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

Il y a :
177 manquants pour la colonnes "âge"
688 manquants pour la colonne "deck"
2 pour les colonnes embarked / embark_town

In [61]:
df["class"].value_counts()

class
Third     491
First     216
Second    184
Name: count, dtype: int64

In [62]:
df["embark_town"].value_counts()

embark_town
Southampton    644
Cherbourg      168
Queenstown      77
Name: count, dtype: int64

Sur 891 personnes, + de 72% des passagers voyagaient pour Southampton

+ de 55% voyageaient en Third Class.

+ de 25% voyageaient en First Class